### ISSUE TITLE
[FEATURE] network-policy resource

### ISSUE THREAD

**Issue by rahearn** [cloudfoundry/terraform-provider-cloudfoundry#83]
### What area do you want to see improved?

terraform provider

### Is your feature request related to a problem? Please describe.

We're using the `cloudfoundry_network_policy` resource of the community provider to allow container-to-container communication on our internal routes. There is no analogous resource in the new provider

### Describe the solution you would like

A new `cloudfoundry_network_policy` resource to be able to fully migrate off of the community provider. The format of the old one is fine with me, but it would also be fine to change it.

Example of the old policy in use:

```
resource "cloudfoundry_network_policy" "backend_routing" {
  policy {
    source_app = data.cloudfoundry_app.source.id
    destination_app = data.cloudfoundry_app.destination.id
    port = "61443"
  }
}
```

**Comment by Dray56** [comment]
The provider uses the [go-cf-client](https://github.com/cloudfoundry/go-cfclient) and [network-policy](https://github.com/cloudfoundry/go-cfclient/issues/376) is not supported by it at the moment. Need to decide whether to contribute there and bring it or develop it as a separate module here in the provider or invoke the existing one used by [community_provider](https://pkg.go.dev/code.cloudfoundry.org/cfnetworking-cli-api@v0.0.0-20190103195135-4b04f26287a6/cfnetworking/cfnetv1).

**Comment by rahearn** [comment]
@Dray56 could this provider use https://github.com/cloudfoundry/policy_client that is mentioned in the issue you linked? I'm concerned about our path forward as the v2 API starts getting turned off and we will no longer be able to continue using the community provider for network policies

**Comment by Dray56** [comment]
@rahearn Yes, that should be the way forward. If you have experience with the client and the time to contribute, please feel free to do so! 😊

**Comment by rahearn** [comment]
@Dray56 thanks - I'll give it a shot this week and see how far I get.
